packages/@lightningjs/ui-components/src/components/ControlRow/ControlRow.d.ts

```diff
@@ -16,34 +16,97 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import type lng from '@lightningjs/core';
-import type { StylePartial } from '../../types/lui';
+import lng from '@lightningjs/core';
+import { StylePartial } from '../../types/lui';
 import TitleRow, { TitleRowStyle } from '../TitleRow';
 
-export type ControlRowStyle = TitleRowStyle & {
+type ControlRowItem = {
+  extraItemSpacing?: number;
+};
+
+type ControlRowStyle = TitleRowStyle & {
   extraItemSpacing: number;
 };
 
-type ControlRowItem = Array<
-  lng.Component.NewPatchTemplate<typeof lng.Component> | lng.Component
->;
+declare namespace ControlRow {
+  export interface TemplateSpec extends TitleRow.TemplateSpec {
+    /**
+     * additional space to be added between controls and content items
+     */
+    extraItemSpacing?: number;
+  }
+}
+export interface TypeConfig extends lng.Component.TypeConfig {
+  SignalMapType: SignalMap;
+}
+/**
+ * The signal is emitted when an item is selected after an index determined by the lazyLoadBuffer property.
+ */
+export type SignalMap = {
+  loadMoreItems(): void;
+};
 
-export default class ControlRow extends TitleRow {
-  leftControls?: ControlRowItem;
-  contentItems?: ControlRowItem;
-  rightControls?: ControlRowItem;
+declare class ControlRow<
+  TemplateSpec extends ControlRow.TemplateSpec = ControlRow.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
+> extends TitleRow<TemplateSpec, TypeConfig> {
+  /**
+   * additional space to be added between controls and content items
+   */
   extraItemSpacing?: number;
+
   get style(): ControlRowStyle;
   set style(v: StylePartial<ControlRowStyle>);
 
-  loadMoreItems(): void;
```

Comment:
we should still have this method in the declaration, but I'm also a little confused as to the purpose of this method